includes/documents/abstract-wcpdf-order-document.php

```diff
@@ -818,8 +818,14 @@ public function has_header_logo() {
 	 * Return logo id
 	 */
 	public function get_header_logo_id() {
-		if ( !empty( $this->settings['header_logo'] ) ) {
-			return apply_filters( 'wpo_wcpdf_header_logo_id', $this->settings['header_logo'], $this );
+		if ( ! empty( $this->settings['header_logo'] ) ) {
+			$header_logo_id = $this->get_settings_text( 'header_logo', '', false );
+
+			if ( empty( $header_logo_id ) ) {
+				$header_logo_id = $this->settings['header_logo'];
```

Comment:
I think `$this->settings['header_logo']` can be an `array`? And in that case this will throw error [here](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/f312b8657ee4ec91a9caa940a8ce62160882a668/includes/documents/abstract-wcpdf-order-document.php#L850)?

Reply:
> I think `$this->settings['header_logo']` can be an `array`? And in that case this will throw error [here](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/f312b8657ee4ec91a9caa940a8ce62160882a668/includes/documents/abstract-wcpdf-order-document.php#L850)?

This should not happen because `$this->get_settings_text( 'header_logo', '', false );` should handle the array version of `$this->settings['header_logo'];`. If that fails, and `$header_logo_id` is still empty, this means that we don't have a language version of logo, and therefore, the value should not be an array.

@alexmigf can you please describe what are the steps you have taken to get the error described above?

Reply:
I see. I'm not able to reproduce the error above again.

Maybe we could remove this part? Since it's handled by the `get_settings_text` function.

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/cf4d8cfdfb921bf8d8590ac82a236ca75a13521a/includes/documents/abstract-wcpdf-order-document.php#L824-L826

Reply:
Actually, after remove the 2 language logos I had, disable WPML and change this branch to `main` I'm now getting this error:

![Screenshot from 2024-04-10 12-15-40](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/assets/533273/a44eca4b-9e90-4984-8732-87069418662c)

After I save a new logo the error is gone.

Reply:
@alexmigf I have removed it, and after testing, I realized I need to make a slight change to the `get_settings_text` method. It should be working correctly now.